src/legilogic_ethereum/ethereum_user.ml

```diff
@@ -611,7 +647,7 @@ let make_pre_transaction ~sender operation ?gas_limit ~value : PreTransaction.t
   if ethereum_user_log then
     Logging.log "ETHUSR: make_pre_transaction gas_limit=%s value=%s" (TokenAmount.to_string gas_limit) (TokenAmount.to_string value);
   (* TODO: The multiplication by 2 is a hack that needs to be addressed *)
-  let gas_limit_n_fold = (TokenAmount.mul (TokenAmount.of_int 2) gas_limit) in
+  let gas_limit_n_fold = (TokenAmount.mul (TokenAmount.of_int 3) gas_limit) in
```

Comment:
Why do we need increase the limit that much, already? I thought you had identified and fixed the problem with gas calculation.

Note that metamask seems to add 50% to the gas estimate and be happy with it. x3 seems a bit excessive.

Reply:
Ok, reverted to 2.